specviz/plugins/unit_change/tests/test_unit_conversion_gui.py

```diff
@@ -1,59 +1,137 @@
-#     assert uc.ui.comboBox_spectral.currentText() == "Custom"
-#
-#     uc.ui.line_custom_spectral.setText("fT")
-#     assert uc.ui.on_accepted() == True
-#
-#
-# def test_custom_units_incorrect(qtbot):
-#     uc = UnitChangeDialog()
-#     uc.show()
-#     qtbot.addWidget(uc)
-#
-#     uc.ui.comboBox_spectral.setCurrentIndex(uc.ui.comboBox_spectral.count() - 1)
-#     assert uc.ui.comboBox_spectral.currentText() == "Custom"
-#
-#     uc.ui.line_custom_spectral.setText("feet")
-#     assert uc.ui.on_accepted() == False
-#
-#
-# def test_accept_works_correctly(qtbot):
-#     uc = UnitChangeDialog()
-#     uc.show()
-#     qtbot.addWidget(uc)
-#
-#     uc.ui.comboBox_spectral.setCurrentIndex(4)
-#     assert uc.ui.comboBox_spectral.currentText() == uc._units_titles[4]
-#
-#     qtbot.mouseClick(uc.ui.buttonBox.button(QDialogButtonBox.Ok), QtCore.Qt.LeftButton)
-#     assert uc.current_units == uc._units_titles[4]
-#
-#
-# def test_cancel_works_correctly(qtbot):
-#     uc = UnitChangeDialog()
-#     uc.show()
-#     qtbot.addWidget(uc)
-#
-#     old_units = uc.current_units
-#
-#     uc.ui.comboBox_spectral.setCurrentIndex(3)
-#     assert uc.ui.comboBox_spectral.currentText() == uc._units_titles[3]
-#
-#     qtbot.mouseClick(uc.ui.buttonBox.button(QDialogButtonBox.Cancel), QtCore.Qt.LeftButton)
-#     assert uc.current_units == old_units
+from qtpy.QtWidgets import QDialogButtonBox
+
+from specviz.plugins.unit_change.unit_change_dialog import UnitChangeDialog
+
+
+def get_ucd(specviz_gui):
+    ucd = UnitChangeDialog(specviz_gui.current_workspace)
+    ucd.show()
+    return ucd
+
+
+def test_spectral_custom_units_correct(specviz_gui, qtbot):
+    """
+    Accept valid custom units
+    """
+    ucd = get_ucd(specviz_gui)
+    qtbot.addWidget(ucd)
+
+    ucd.ui.comboBox_spectral.setCurrentIndex(0)
+    assert ucd.ui.comboBox_spectral.currentText() == ucd.spectral_axis_unit_equivalencies_titles[0]
+
+    ucd.ui.comboBox_spectral.setCurrentIndex(ucd.ui.comboBox_spectral.count()-1)
+    assert ucd.ui.comboBox_spectral.currentText() == "Custom"
+
+    ucd.ui.line_custom_spectral.setText("m")
+    assert ucd.on_accepted() == True
+
+
+def test_spectral_custom_units_incorrect(specviz_gui, qtbot):
+    """
+    Reject invalid custom units
+    """
+    ucd = get_ucd(specviz_gui)
+    qtbot.addWidget(ucd)
+
+    ucd.ui.comboBox_spectral.setCurrentIndex(ucd.ui.comboBox_spectral.count() - 1)
+    assert ucd.ui.comboBox_spectral.currentText() == "Custom"
+
+    ucd.ui.line_custom_spectral.setText("feet")
+    assert ucd.on_accepted() == False
+
+
+def test_spectral_accept_works_correctly(specviz_gui, qtbot):
+    """
+    Accept works for units in combobox
+    """
+    ucd = get_ucd(specviz_gui)
+    qtbot.addWidget(ucd)
+    
+    if ucd.ui.comboBox_spectral.count() > 4:
+        ucd.ui.comboBox_spectral.setCurrentIndex(4)
+        assert ucd.ui.comboBox_spectral.currentText() == ucd.spectral_axis_unit_equivalencies_titles[4]
+    else:
```

Comment:
Under what conditions would either of these be executed? I'm a little uncomfortable with tests that could have different results on different runs.

Reply:
Agree with @drdavella: this should only be testing a single, expected value.